expertise/utils/utils.py

```diff
@@ -510,6 +511,65 @@ def aggregate_by_group(config):
             for archive_member, score in archive_scores.items():
                 csvwriter.writerow([archive_member, submission_member, score])
 
+def cache_group(model_name, group_id, work_dir, pub_dir, dataset_dir):
+    # Create a cache directory for the current group's embedding (SPECTER+MFR only)
+    if isinstance(group_id, list):
+        assert len(group_id) == 1, f"Cache is only enabled for 1 group, {len(group_id)} found"
+        group_id = group_id[0]
+    group_dir = group_id.replace('/', '_')
+    cache_dir = f"../groups/{model_name}/{group_dir}"
+    if not os.path.isdir(cache_dir):
+        os.makedirs(cache_dir)
+
+    # Build + dump archive mapping
+    archive_map = {}
+    archive_dir = f"{dataset_dir}/archives"
+    member_fnames = os.scandir(archive_dir)
+    for fname in member_fnames:
+        member_id = fname.name[:-6]
+        archive_map[member_id] = []
+
+        # Retrieve store publications
+        with open(f"{archive_dir}/{fname.name}") as member_file:
+            pub_list = list(member_file)
+        for pub_string in pub_list:
+            pub_dict = json.loads(pub_string)
+            archive_map[member_id].append(pub_dict.get('id', ''))
+    with open(f"{cache_dir}/archives.json", 'w') as f:
+        json.dump(archive_map, f)
+
+    # Copy work files - MFR is stored in the work_dir, SPECTER is stored in pub2vec
+    shutil.copytree(f"{work_dir}/mfr", f"{cache_dir}/work")
+    shutil.copy2(f"{pub_dir}/pub2vec.jsonl", f"{cache_dir}")
+    
+def from_cache(model_name, group_id, work_dir, pub_dir, dataset_dir):
```

Comment:
do you still need this? I thought that all the caching was going to happen through Redis. Why do you need to cache the files?

Reply:
They are not used anymore, will remove these.